src/main/java/com/epam/container/TransportList.java

```diff
@@ -0,0 +1,388 @@
+        return new TransportListIterator<>(predicate);
+    }
+
+    /**
+     * Returns an array containing all the elements in this list in proper sequence (from first to last element)
+     */
+    @Override
+    public Object[] toArray() {
+        return Arrays.copyOf(containerOfTransports, size);
+    }
+
+    /**
+     * Returns an array containing all the elements in this list in proper sequence (from first to last element);
+     * the runtime type of the returned array is that of the specified array.
+     */
+    @Override
+    public <T> T[] toArray(T[] ts) {
+        return (T[]) toArray();
+    }
+
+    /**
+     * Appends the specified element to the end of this list
+     *
+     * @param transport - element to be appended to this list
+     * @return true (as specified by Collection.add(E))
+     */
+    @Override
+    public boolean add(Transport transport) {
+        resize();
+        containerOfTransports[size++] = transport;
+        return true;
+    }
+
+    /**
+     * Removes the first occurrence of the specified element from this list, if it is present
+     *
+     * @param o - element to be removed from this list, if present
+     * @return true if this list contained the specified element
+     */
+    @Override
+    public boolean remove(Object o) {
+        int index = indexOf(o);
+        if (index >= 0 && index < size) {
+            System.arraycopy(containerOfTransports, index + 1, containerOfTransports, index, size - index - 1);
+            size--;
+            containerOfTransports[size] = null;
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Returns true if this list contains all the elements of the specified collection
+     *
+     * @param collection - collection to be checked for containment in this list
+     * @return true if this list contains all the elements of the specified collection
+     */
+    @Override
+    public boolean containsAll(Collection<?> collection) {
+        for (Object o : collection) {
+            if (!contains(o)) return false;
+        }
+        return true;
+    }
+
+    /**
+     * Appends all the elements in the specified collection to the end of this list
+     *
+     * @param collection - collection containing elements to be added to this list
+     * @return true if this list changed as a result of the call
+     */
+    @Override
+    public boolean addAll(Collection<? extends Transport> collection) {
+        collection.forEach(this::add);
+        return true;
+    }
+
+    /**
+     * @param i          - index at which to insert the first element from the specified collection
+     * @param collection - collection containing elements to be added to this list
+     * @return true if this list changed as a result of the call
+     */
+    @Override
+    public boolean addAll(int i, Collection<? extends Transport> collection) {
+        collection.forEach(obj -> add(i, obj));
```

Comment:
in this case u will lose an order of elements, u need to save order like
[1, 2, 3, 4] - old arr,
[7,9] new collection and index 2
-> [1, 2, 7, 9, 3, 4]